presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -611,7 +616,7 @@ private void verifyNoOrderByReferencesToOutputColumns(Node node)
         getReferencesToScope(node, analysis, orderByScope.get())
                 .findFirst()
                 .ifPresent(expression -> {
-                    throw new SemanticException(REFERENCE_TO_OUTPUT_ATTRIBUTE_WITHIN_ORDER_BY_AGGREGATION, expression, "Invalid reference to output projection attribute from ORDER BY aggregation");
+                    throw new SemanticException(REFERENCE_TO_OUTPUT_ATTRIBUTE_WITHIN_ORDER_BY_AGGREGATION, expression, "Invalid reference to output projection attribute from ORDER BY expression");
```

Comment:
I'm a little hesitant to make this change but I did it anyway. Having two error codes that are each used only once seems like overkill. In a previous review Raghav and Martin told me to remove grouping related error codes.

Also returning a boolean won't work since you lose the expression that caused the error. You'll instead have to return an Optional. I decided to make the method take as arguments the error code and the error message.